### DISCUSSION TITLE
Made translation of week programs labels flexible

### DISCUSSION THREAD

**Discussion by prodigy7** [SukramJ/aiohomematic#1767] [Ideas]
Hey there!

Currently the labels of the different week programs are hardcodes:

https://github.com/danielperna84/custom_homematic/blob/24d11c2e4f90e431208c7394a15a7267abe526ba/custom_components/homematicip_local/translations/de.json#L207-L212

It would be nice, if there would be a way, set a optional own custom label. Background in my case is, that I want to use week program "work" for weeks where I'm working outside, "holiday" if I'm at home. The current labels "WP 1", "WP 2" and so on are not really intuitive.

**Comment by SukramJ** [comment]
These labels are called translations, and the mechanism is provided by HA.
If you want to customize translations, then you should start a discussion at the [HA archtecture repo](https://github.com/home-assistant/architecture/discussions).

Other option might be to modify the card use CSS or create a climate template, but this is not part of the integration.

**Comment by prodigy7** [comment]
Opened a new thread here. For everyone want follow, here the link: https://github.com/home-assistant/architecture/discussions/1156

**Comment by SukramJ** [comment]
Referencing a custom_component is not a good decision in a core repository!
The impact must always be explained in reference to a core integration or their model.
